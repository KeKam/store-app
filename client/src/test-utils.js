import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';

import rootReducer from './redux/root-reducer';

// const render = (
//   component,
//   {
//     initialState,
//     store = createStore(rootReducer, initialState),
//     ...renderOptions
//   } = {}
// ) => {
//   return {
//     ...rtlRender(<Provider store={store}>{component}</Provider>),
//     store,
//   };
//   // const Wrapper = ({ children }) => {
//   //   return <Provider store={store}>{children}</Provider>;
//   // };
//   // return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
// };

const render = (component, { initialState } = {}) => {
  const actions = [];
  const observerMiddleware = () => (next) => (action) => {
    actions.push(action);
    return next(action);
  };
  const store = createStore(
    rootReducer,
    initialState,
    applyMiddleware(observerMiddleware)
  );
  const utils = {
    dispatch(action) {
      return store.dispatch(action);
    },
    getDispatchedActions() {
      return actions;
    },
    getState() {
      return store.getState();
    },
  };
  return {
    ...rtlRender(<Provider store={store}>{component}</Provider>),
    ...utils,
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { render };
