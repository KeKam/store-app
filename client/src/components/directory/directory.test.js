import React from 'react';
import '@testing-library/jest-dom';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

import { render, screen } from '../../test-utils';
import Directory from './directory';

const history = createMemoryHistory();

describe('Directory component', () => {
  it('should render Directory component', () => {
    const mockState = {
      sections: [
        {
          title: 'SPRING',
          imageUrl: 'www.testImage.com',
          id: 1,
          linkUrl: 'collection/spring',
        },
      ],
    };

    const { asFragment } = render(
      <Router history={history}>
        <Directory />
      </Router>,
      { initialState: { directory: mockState } }
    );

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByTestId('item')).toBeInTheDocument();
    expect(screen.getByTestId('image')).toBeInTheDocument();
    expect(screen.getByTestId('title')).toHaveTextContent('SPRING');
    expect(screen.getByText('SEE COLLECTION')).toBeInTheDocument();
  });
});
