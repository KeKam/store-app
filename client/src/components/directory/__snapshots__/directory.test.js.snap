// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory component should render Directory component 1`] = `
<DocumentFragment>
  <div
    class="sc-EHOje fpJsVf"
  >
    <div
      class="sc-bdVaJa hJLFdt"
      data-testid="item"
    >
      <div
        class="sc-bwzfXH klmmtG"
        data-testid="image"
      />
      <div
        class="sc-htpNat fMwJqu"
      >
        <div
          class="sc-bxivhb hQYITJ"
          data-testid="title"
        >
          SPRING
        </div>
        <div
          class="sc-ifAKCX jbyhsX"
        >
          SEE COLLECTION
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
